gnucash/import-export/import-backend.c

```diff
@@ -1145,9 +1077,27 @@ static gint check_trans_online_id(Transaction *trans1, void *user_data)
     }
 }
 
+static gint collect_trans_online_id(Transaction *trans, void *user_data)
+{
+    Split *split;
+    GHashTable *id_hash = user_data;
+    int i=0;
+    
+    const gchar* online_id = gnc_import_get_trans_online_id (trans);
+    if (online_id)
+        g_hash_table_add (id_hash, (void*) online_id);
+
+    for (GList *splits = xaccTransGetSplitList (trans); splits; splits = splits->next)
+    {
+        if (gnc_import_split_has_online_id (splits->data))
+            g_hash_table_add(id_hash, (void*) gnc_import_get_split_online_id (splits->data));
+    }
+    return 0;
+}
+
```

Comment:
I am aware this function `collect_trans_online_id` was moved away, but this is the first occurrence of the `g_hash_table_add` usage. @jeanlaroche would you be able to review the very recent commit 579ba443172fde921fd6526e37a0a30be7d2da5e which has changed `g_hash_table_add (hash, id)` to `g_hash_table_insert (hash, id, 1)` -- the reason is that `id` is a newly allocated string and and needs to be freed after use, and if we use `id` as both key and value in the hash table, then I'm not sure whether to free in the key or value.

Reply:
First, I don't think there's any issue with using g_hash_table_insert (hash, id, 1), it might be slightly less efficient memorywise but I can't imagine that being a problem.
Second, if you use g_hash_table_add , I imagine you'd just free the key when the hash table is destroyed.

Reply:
Thanks for clarification. Agree it probably doesn't matter that much, and now there will be fewer leaks.

Reply:
From a leak perspective `g_hash_table_add` has a wrinkle: It's a wrapper of `g_hash_table_replace` that, like its name suggests, discards the existing key and value and inserts the new one. To prevent leaks it's therefore important to set the `key_destroy_func` or the `value_destroy_func` when creating the hash table so that one or the other gets cleaned up; there's no other way to get the old key/value to free it.